spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryProperties.java

```diff
@@ -219,13 +241,41 @@ public ConsulDiscoveryProperties(InetUtils inetUtils) {
 
 	/**
 	 * @param serviceId The service who's filtering tag is being looked up
-	 * @return The tag the given service id should be filtered by, or null.
+	 * @return The tags the given service id should be filtered by, or null.
+	 * @deprecated instead use {@link #getQueryTagsForService(String)}
 	 */
+	@Deprecated
 	public String getQueryTagForService(String serviceId) {
 		String tag = this.serverListQueryTags.get(serviceId);
 		return tag != null ? tag : this.defaultQueryTag;
 	}
 
+	/**
+	 * Gets the list of tags to use when looking up the instances for a particular
+	 * service. If the service has an entry in {@link #serviceInstanceQueryTags} those
+	 * will be used. Otherwise the content of {@link #defaultServiceInstanceQueryTags}
+	 * will be used - defaulting to an empty list if neither has entries.
+	 * @param serviceId the service whose instances are being looked up
+	 * @return non-null list of tags to filter the service instances by (empty if no tags
+	 * configured)
+	 */
+	public List<String> getQueryTagsForService(String serviceId) {
+		// Use specific config entry if available
+		if (this.serviceInstanceQueryTags.containsKey(serviceId)) {
+			return Collections.unmodifiableList(this.serviceInstanceQueryTags.get(serviceId));
+		}
+
+		// Honor use of deprecated property while giving tip to consuming application
```

Comment:
We honor the deprecated `defaultQueryTag` so that if consuming apps were relying on that to feed into the services query, it still will do that. 

I chose to not honor the elder `serverListQueryTags` since there is no internal use of it. IOW - this will not break anyone that is setting that property. The only way that property has any impact is if consumers are also using the deprecated singular `getQueryTagForService` for some other external use. When/if consumers that are setting `serverListQueryTags` decide to move off of it they can also move off of the deprecated singular accessor `getQueryTagForService`.